src/Basic/Validation.cs

```diff
@@ -0,0 +1,24 @@
+using AslHelp.Data.AutoSplitter;
+
+namespace AslHelp;
+
+public static class Validation
+{
+    public static void AssertNotNull(string name, dynamic value)
+    {
+        if (value == null)
+        {
+            throw new ArgumentNullException($"{name} may not be null.");
+        }
+    }
+
+    public static void AssertAction(string name, string action)
```

Comment:
call this `InAction`. use `CallerMemberNameAttribute` for automatic caller name inference.